### PR TITLE
google: update workflow versions and instructions

### PR DESCRIPTION
**Pull request by sethvargo**
## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [x] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [x] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [x] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

### CODE REVIEW

File: deployments/google-cloudrun-docker.yml [hunk 1]
```diff
@@ -1,113 +1,95 @@
-# This workflow build and push a Docker container to Google Artifact Registry and deploy it on Cloud Run when a commit is pushed to the $default-branch branch
-#
-# Overview:
-#
-# 1. Authenticate to Google Cloud
-# 2. Authenticate Docker to Artifact Registry
-# 3. Build a docker container
-# 4. Publish it to Google Artifact Registry
-# 5. Deploy it to Cloud Run
+# This workflow build and push a Docker container to Google Artifact Registry
```
Comment (janbrasna) [inline deployments/google-cloudrun-docker.yml:1]:
```suggestion
# This workflow will build and push a Docker container to Google Artifact Registry
```

**Review by alexisabril — APPROVED** [review]
Thank you for the PR @sethvargo!

File: deployments/google-cloudrun-docker.yml [hunk 2]
```diff
@@ -1,113 +1,95 @@
...
     permissions:
       contents: 'read'
       id-token: 'write'
 
-    runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - name: 'Checkout'
+        uses: 'actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332' # actions/checkout@v4
```
Comment (jsoref) [inline deployments/google-cloudrun-docker.yml:53]: @alexisabril I thought that github's guidance said not to sha pin `actions/*@`.

File: deployments/google-cloudrun-docker.yml (hunk 1 above)
Comment (FKNCOIN) [reply deployments/google-cloudrun-docker.yml:1]: K

File: deployments/google-cloudrun-docker.yml (hunk 2 above)
Comment (965022-oss) [reply deployments/google-cloudrun-docker.yml:53]: Merge #2478 Branch: main 965022-oss-patch3/SHA f4dbb8424c

**Pull request merged**